### PR TITLE
feat(gatsby-plugin-postcss): added css-loader options

### PR DESCRIPTION
**Pull request by magicspon**
## Description

This commit allows users to define some css-loader options, such as `localIdentName` 

```javascript
{
	resolve: `gatsby-plugin-postcss`,
	options: {
		postCssPlugins: plugins,
		cssOptions: {
			localIdentName: 'mod-[hash:base64:8]'
		}
	}
}
``` 

`cssOptions` is the new bit.

## Related Issues

Some of my class names were getting stupidly long!

### CODE REVIEW

**Comment by magicspon** [comment]
yup, that makes sense, i've changed it to cssLoaderOptions.

thanks

File: packages/gatsby-plugin-postcss/README.md [hunk 1]
```diff
@@ -53,3 +53,19 @@ module.exports = () => ({
...
 })
 ```
+
+If you need to override the default options passed into [`css-loader`](https://github.com/webpack-contrib/css-loader)
+
+```javascript
+// in gatsby-config.js
+plugins: [
+  {
+    resolve: `gatsby-plugin-less`,
```
Comment (pieh) [inline packages/gatsby-plugin-postcss/README.md]:
```suggestion
    resolve: `gatsby-plugin-postcss`,
```
:)

**Comment by pieh** [comment]
We also need to update jest snapshots as seems slight order change of options make them fail

**Comment by mbifulco** [comment]
Hey there - anything I can do to help with this PR? It looks like this would be helpful for a project I'm working on.

My use case:
- I'm setting up Gatsby and Storybookjs in the same project.  Right now, Gatsby and Storybook have both chosen to feed opposing options into the `css-loader` `camelCase` option, which means the same `styles.module.css` file gets compiled down to a `camelCase` based dictionary in one project, and a `skewer-case` dictionary in the other:

# In other words...
```js
import classes from './Footer.module.css';

console.log('footer classes', classes);
```

This results in 
## Camel Case from gatsby:
image

## Skewer-case from storybook:
image

This essentially renders storybook and gatsby incompatible for my purposes.  I'm hoping this PR would help me address that!

**Comment by wardpeet** [comment]
i'll get this one merged tomorrow @mbifulco

**Comment by magicspon** [comment]
heyo... sorry i've not had a chance to take another look at this... let me know if there is anything i can do to help. 

cheers

**Comment by mbifulco** [comment]
I hate to be a pain, but it looks like failing tests may be holding this up.  Is that jest missing a snapshot update?

**Comment by wardpeet** [comment]
sorry fell sick yesterday, updated the snapshots 🤞

**Comment by mbifulco** [comment]
Ah sorry to hear you were under the weather.  I hope you're back at full steam now!

Thank you, @wardpeet  @magicspon @pieh! 🙏

**Comment by mbifulco** [comment]
Just coming back to say that this absolutely did the trick for me.  Thanks, y'all! This is the postcss config I needed:

```es6
    {
      resolve: `gatsby-plugin-postcss`,
      options: {
        cssLoaderOptions: {
          camelCase: true,
        },
      },
    },
```

and this is a screenshot of thegymnasium.com's footer running simultaneously in storybook and `gatsby develop` 👍 Sweet!

image
